File: BaseLibrary/communication/command/equipment/GetGunsAnswerCommand.cs
using System.Collections.Generic;
using BaseLibrary.equipment;

namespace BaseLibrary.communication.command.equipment {
    public class GetGunsAnswerCommand : AEquipmentCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        /// <summary>
        /// Available guns to buy.
        /// </summary>
        /// <seealso cref="Gun"/>
        public Gun[] GUNS { get; private set; }

        public GetGunsAnswerCommand(Gun[] guns)
            : base() {
				GUNS = guns;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/equipment/GetGunsAnswerCommand.java
package BaseLibrary.communication.command.equipment;

import BaseLibrary.communication.protocol.AProtocol;
import BaseLibrary.equipment.Gun;

public class GetGunsAnswerCommand extends AEquipmentCommand {

	private static final String NAME = "GUNS_ANSWER";

	static {
		AProtocol.registerForDeserialize(NAME, GetGunsAnswerCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	private Gun[] guns;

	public GetGunsAnswerCommand(Gun[] guns) {
		setGuns(guns);
	}

	/**
	 * Available guns to buy.
	 * 
	 * @see Gun
	 */
	public Gun[] getGuns() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return guns;
	}

	private void setGuns(Gun[] guns) {
		this.guns = guns;
		pending = false;
	}
}
